### PR TITLE
Test/insufficient funds for zkapps

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Adding new tests to the `zkapp-payments` test, the test create transfers that over the balance of the account and tests that the update is not successful with the `failure-status` (to be specific, the failure_status should be `Overlow`).

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

File: src/lib/transaction_snark/test/snapp_payments/snapp_payments.ml [hunk 1]
```diff
@@ -190,4 +190,78 @@ let%test_module "Snapp payments tests" =
...
+              let logger = Logger.create () in
+              [%log info] "failure_status"
+                ~metadata:
+                  [ ( "failure_status"
+                    , `List
+                        (List.map failure_status
+                           ~f:Transaction_status.Failure.to_yojson) )
+                  ] ;
+              assert (
+                List.mem failure_status Transaction_status.Failure.Overflow
```
Comment (psteckler) [inline src/lib/transaction_snark/test/snapp_payments/snapp_payments.ml:265]: Should there be other failures? If there's only one failure, maybe look at the whole list.

File: src/lib/transaction_snark/test/snapp_payments/snapp_payments.ml [hunk 2]
```diff
@@ -190,4 +190,78 @@ let%test_module "Snapp payments tests" =
                     (module Ledger.Ledger_inner)
                     init_ledger ledger ;
                   U.apply_parties_with_merges ledger [ parties ])))
+
```
Comment (psteckler) [inline src/lib/transaction_snark/test/snapp_payments/snapp_payments.ml:193]: should the directory and file be renamed with a `zkapp` prefix?

File: src/lib/transaction_snark/test/snapp_payments/snapp_payments.ml (hunk 1 above)
Comment (ghost-not-in-the-shell) [reply src/lib/transaction_snark/test/snapp_payments/snapp_payments.ml:265]: There's another failure: `Invalid_fee_excess`

File: src/lib/transaction_snark/test/util.ml [hunk 3]
```diff
@@ -47,6 +47,8 @@ let genesis_state_body =
   in
   compile_time_genesis.data |> Mina_state.Protocol_state.body
 
+let state_view = Mina_state.Protocol_state.Body.view genesis_state_body
```
Comment (psteckler) [inline src/lib/transaction_snark/test/util.ml:50]: maybe `genesis_state_view` would be a better name

File: src/lib/transaction_snark/test/zkapp_payments/zkapp_payments.ml [hunk 4]
```diff
@@ -186,4 +189,81 @@ let%test_module "Snapp payments tests" =
                     (module Ledger.Ledger_inner)
                     init_ledger ledger ;
                   U.check_parties_with_merges_exn ledger [ parties ])))
+
+    let%test_unit "snapps payments failed due to insufficient funds" =
```
Comment (psteckler) [inline src/lib/transaction_snark/test/zkapp_payments/zkapp_payments.ml:193]: `zkapps` instead of `snapps`?

**Review by psteckler — CHANGES_REQUESTED** [review]
a couple of nits about names of things

**Review by psteckler — APPROVED** [review]

**Pull request merged**